src/include/migraphx/op/broadcast.hpp

```diff
@@ -49,8 +49,6 @@ struct broadcast
         if(std::all_of(
                broadcast_lens.cbegin(), broadcast_lens.cend(), [&](auto x) { return x == 1; }))
         {
-            if(axis != 0)
-                MIGRAPHX_THROW("BROADCAST: when broadcasting tensor of size 1, axis should be 0");
```

Comment:
Can you add a unit test for this?

Reply:
We have unit tests for this if statement under `op_shape_test.cpp`. I also added a scalar test for broadcast in tf.

Reply:
There isnt a test for non zero axis in op_shape_test.cpp, only for when axis is 0.

Reply:
Added

Reply:
We may need to add a check that input either is a scalar or contains only one element? Then the shape unit test there can still pass.